### ISSUE TITLE
Third parameter does not seem to work properly

### ISSUE THREAD

**Issue by iimog** [ugexe/Text--Levenshtein--XS#10]
I use ```Text::Levenshtein::XS``` on Ubuntu 15.04 (64bit) installed via CPAN:
```
cpan[1]> i Text::Levenshtein::XS
Going to read '/home/markus/.cpan/Metadata'
  Database was generated on Thu, 27 Aug 2015 15:41:02 GMT
Module id = Text::Levenshtein::XS
    CPAN_USERID  UGEXE (Nick Logan <ugexe@cpan.org>)
    CPAN_VERSION 0.43
    CPAN_FILE    U/UG/UGEXE/Text-Levenshtein-XS-0.43.tar.gz
    UPLOAD_DATE  2014-10-15
    MANPAGE      Text::Levenshtein::XS - Calculate edit distance based on insertion, deletion, substitution, and transposition
    INST_FILE    /home/markus/perl5/lib/perl5/x86_64-linux-gnu-thread-multi/Text/Levenshtein/XS.pm
    INST_VERSION 0.43
```
When passing a third parameter to the ```distance``` function there are two kinds of unexpected behaviour:
1) distance is smaller than cut-off and the result is still not reported, eg:
   ```{perl}
   distance('ACGTAG', 'AGTAG', 2) # should be 1 is undef 
   ```
2) distance is larger than cut-off and the result is still returned, eg:
   ```{perl}
   distance('ACGTAG', 'TCGG', 2) # is 3 should be undef
   ```
Unfortunately I could not verify that this problem still exists in version 0.46 on GitHub as I was unable to compile.

**Comment by ugexe** [comment]
This should be fixed in v0.501

https://github.com/ugexe/Text--Levenshtein--XS/blob/master/t/02_distance.t#L74-L75

**Comment by iimog** [comment]
Nice! Thanks for the quick fix.
